File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        a = {}
        b = {}

        for i in s :
            if i in a :
                a[i] += 1
            else :
                a[i] = 1
        for i in t :
            if i in b :
                b[i] += 1
            else :
                b[i] = 1

        return a==b

File: Data Structures & Algorithms/max-water-container/submission-3.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int maxv = 0;
        int l = 0;
        int r = heights.size() -1;
        while (l < r) {
            int vol = (min(heights[l], heights[r]) * (r-l));
            maxv = max(maxv, vol);
            if (heights[l] > heights[r]) {
                --r;
            } else if (heights[l] < heights[r]) {
                ++l;
            } else {
                --r;
            }
        }
        return maxv;
    }
};
